### PR TITLE
feat: Update asserts to use new winter23 Assert class - Integration Recipes

### PR DESCRIPTION
**Pull request by msrivastav13**
### What does this PR do?

Update asserts to use new winter23 Assert class. The new Assert class is really elegant compared to the older System methods.

### What issues does this PR fix or reference?

#<Insert GitHub Issue>

## The PR fulfills these requirements:

[ ] Tests for the proposed changes have been added/updated.
[ ] Code linting and formatting was performed.

### Functionality Before

<insert gif and/or summary>

### Functionality After

<insert gif and/or summary>

### CODE REVIEW

File: force-app/tests/Integration Recipes/CalloutRecipes_Tests.cls [hunk 1]
```diff
@@ -139,7 +139,7 @@ private inherited sharing class CalloutRecipes_Tests {
         Integer statusCode = callouts.httpDeleteCalloutToSecondOrg(contact.Id);
         Test.stopTest();
 
-        System.assert(statusCode == 200, 'Expected a successful delete');
+        System.Assert.isTrue(statusCode == 200, 'Expected a successful delete');
```
Comment (pozil) [inline force-app/tests/Integration Recipes/CalloutRecipes_Tests.cls:142]: Change to `areEqual`

Comment (pozil) [reply force-app/tests/Integration Recipes/CalloutRecipes_Tests.cls:142]: I won't spam you with comments on this file but this remark applies to all `statusCode` tests further down.

File: force-app/tests/Integration Recipes/CalloutRecipes_Tests.cls [hunk 2]
```diff
@@ -336,11 +357,11 @@ private inherited sharing class CalloutRecipes_Tests {
         Map<String, Object> responseMap = callouts.httpCalloutWithUntypedResponse();
         Test.stopTest();
 
-        System.assert(
+        System.Assert.isTrue(
```
Comment (pozil) [inline force-app/tests/Integration Recipes/CalloutRecipes_Tests.cls:360]: Change to `areEqual`

File: force-app/tests/Integration Recipes/CalloutRecipes_Tests.cls [hunk 3]
```diff
@@ -336,11 +357,11 @@ private inherited sharing class CalloutRecipes_Tests {
...
             responseMap.get('Name') == 'Universal Containers',
             'Expected to find Universal Containers as name in the response'
         );
-        System.assert(
+        System.Assert.isTrue(
```
Comment (pozil) [inline force-app/tests/Integration Recipes/CalloutRecipes_Tests.cls:364]: Change to `areEqual`

File: force-app/tests/Integration Recipes/ApiServiceRecipes_Tests.cls [hunk 4]
```diff
@@ -6,7 +6,7 @@ private class ApiServiceRecipes_Tests {
         ApiServiceRecipes api = new ApiServiceRecipes();
         Test.stopTest();
 
-        System.assert(
+        System.Assert.isTrue(
```
[the remark this answers is not in the document]
Comment (msrivastav13) [reply force-app/tests/Integration Recipes/ApiServiceRecipes_Tests.cls]: done!

File: force-app/tests/Integration Recipes/CalloutRecipes_Tests.cls [hunk 5]
```diff
@@ -90,7 +90,7 @@ private inherited sharing class CalloutRecipes_Tests {
         List<Account> calloutResponse = x.httpGetCalloutToSecondOrg();
         Test.stopTest();
 
-        System.assert(
+        System.Assert.isTrue(
```
[the remark this answers is not in the document]
Comment (msrivastav13) [reply force-app/tests/Integration Recipes/CalloutRecipes_Tests.cls]: done!

File: force-app/tests/Integration Recipes/CalloutRecipes_Tests.cls (hunk 3 above)
Comment (msrivastav13) [reply force-app/tests/Integration Recipes/CalloutRecipes_Tests.cls]: done!

**Review by pozil — APPROVED** [review]
All good, thanks.

**Pull request merged**